core/src/components/button/button.tsx

```diff
@@ -124,12 +124,12 @@ export class Button implements ComponentInterface, AnchorInterface, ButtonInterf
   /**
    * Emitted when the button has focus.
    */
-  @Event() ionFocus!: EventEmitter<void>;
+  @Event({ composed: false }) ionFocus!: EventEmitter<void>;
 
   /**
    * Emitted when the button loses focus.
    */
-  @Event() ionBlur!: EventEmitter<void>;
+  @Event({ composed: false }) ionBlur!: EventEmitter<void>;
```

Comment:
Do we still need these?

Reply:
Yeah, they prevent duplicate blur/focus events from firing on non-Safari browsers. (Explained in the second "new behavior" bullet)

Reply:
I pushed a commit that should resolve this issue without changing `composed` on `ion-button` events. My main concern with the previous approach is that this is a breaking change to the behavior of a public API, and we try to make those only when absolutely necessary.

The code is at https://github.com/ionic-team/ionic-framework/pull/23980/commits/4eddaeb97b85055c5bdd57ff385d83872aeda98e - let me know your thoughts.

Reply:
Aha, yeah that looks great 👍 I initially had the thought of using `stopPropagation`, but couldn't think of a clean way to do it that wouldn't involve adding listeners to a bunch of different spots. I'll update the PR description, and keep that in mind!